run_service.sh

```diff
@@ -156,32 +247,16 @@ then
       awk '{ print substr( $0, '$pkey_start_position', length($0) - '$pkey_start_position' ) }')
     echo "Your agent instance's autogenerated public address: $agent_address"
     echo -n "$agent_address" > "../$agent_address_path"
+    echo ""
 
     # Check balances
     agent_balance=0
     operator_balance=0
     suggested_amount=50000000000000000
-    until [[ $($PYTHON_CMD -c "print($agent_balance > ($suggested_amount-1))") == "True" && $($PYTHON_CMD -c "print($operator_balance > ($suggested_amount-1))") == "True" ]];
-    do
-        echo "Agent instance's address: $agent_address. Balance: $agent_balance WEI."
-        echo "Operator's address: $operator_address. Balance: $operator_balance WEI."
-        echo "Both of the addresses need to be funded to cover gas costs."
-        echo "Please fund them with at least 0.05 xDAI each to continue."
-        echo "Checking again in 10s..."
-        sleep 10
-        agent_balance=$(curl -s -S -X POST \
-          -H "Content-Type: application/json" \
-          --data '{"jsonrpc":"2.0","method":"eth_getBalance","params":["'"$agent_address"'","latest"],"id":1}' "$rpc" | \
-          $PYTHON_CMD -c "import sys, json; print(json.load(sys.stdin)['result'])")
-        operator_balance=$(curl -s -S -X POST \
-          -H "Content-Type: application/json" \
-          --data '{"jsonrpc":"2.0","method":"eth_getBalance","params":["'"$operator_address"'","latest"],"id":1}' "$rpc" | \
-          $PYTHON_CMD -c "import sys, json; print(json.load(sys.stdin)['result'])")
-        agent_balance=$((16#${agent_balance#??}))
-        operator_balance=$((16#${operator_balance#??}))
-    done
 
-    echo "Minting your service on Gnosis chain..."
+    ensure_minimum_balance $operator_address $suggested_amount "Please, fund the operator's address with at least $(wei_to_dai $suggested_amount) DAI."
+
```

Comment:
@Adamantios Here I removed the check for the minimum balance of the agent instance address.